libs/core/src/lib/fixed-card-layout/fixed-card-layout.component.ts

```diff
@@ -0,0 +1,214 @@
+import { CdkDragDrop, CdkDropList, moveItemInArray } from '@angular/cdk/drag-drop';
+import {
+    AfterContentInit,
+    AfterViewChecked,
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+import { fromEvent, Subject, Subscription } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { RtlService } from '../utils/services/rtl.service';
+
+const CARD_MINIMUM_WIDTH = 320; // in px; 20rem max card size
+const CARD_GAP_WIDTH = 16; // gap=1rem==16px
+
+@Directive({ selector: '[fdCardDef]' })
+export class CardDefinitionDirective {
+    constructor(public template: TemplateRef<any>) {}
+}
+
+class Layout {
+    constructor(public numberOfColumns: number, public screenSize: number) {}
+}
+
+class CardDropped {
+    constructor(public container: CdkDropList, public previousIndex: number, public currentIndex: number) {}
+}
+
+type CardColumn = CardDefinitionDirective[];
+
+@Component({
+    selector: 'fd-fixed-card-layout',
+    templateUrl: './fixed-card-layout.component.html',
+    styleUrls: ['./fixed-card-layout.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
Let's remove styles encapsulation on this component

Reply:
included encapsulation: ViewEncapsulation.None